### DISCUSSION TITLE
Diagram as a Code

### DISCUSSION THREAD

**Discussion by Anton-Aracor** [excalidraw/excalidraw#8889] [Q&A]
Hello guys.

I want to use something for Diagram as a Code, but that looks good.

I want to create human readable text file with good syntax highlight in VS Code, with the preview as an image available.

Then I want to add GitHub Action that will convert Diagram as a Code file to the actual image, that will be placed on the README.md file of our repository on GitHub.

Is is possible to do right now for Excalidraw?

**Comment by Mrazator** [comment]
> I want to create human readable text file with good syntax highlight in VS Code, with the preview as an image available.

As of now, there is no diagram as a code for generic diagrams.

The closest thing is likely text-to-diagram https://x.com/excalidraw/status/1729163853546868802, but that's just for few types of mermaid diagrams.

An alternative might be using the excalidraw vscode extension and storing the metadata inside `.excalidraw.png` file to be re-opened later.

**Reply by Anton-Aracor** [reply]
It is sad. As the tool have really nice look. And other tools in the market that generate images from the code are a bit ugly.

Is it any active work towards this direction?

**Reply by Mrazator** [reply]
> Is it any active work towards this direction?

No, not at the moment
